isis/src/hayabusa/apps/amicacal/AmicaCalUtils.h

```diff
@@ -96,42 +97,54 @@ static void loadNaifTiming() {
  *
  * @return @b double Distance in AU between Sun and observed body.
  */
-static bool sunDistanceAU(const QString &scStartTime,
+static bool sunDistanceAU(Cube *iCube,
+                          const QString &scStartTime,
                           const QString &target,
                           double &sunDist) {
 
   //  Ensure NAIF kernels are loaded
-  loadNaifTiming();
-  sunDist = 1.0;
-
-  //  Determine if the target is a valid NAIF target
-  SpiceInt tcode;
-  SpiceBoolean found;
-  bodn2c_c(target.toLatin1().data(), &tcode, &found);
-
-  if (!found) return (false);
-
-  //  Convert starttime to et
-  double obsStartTime;
-  scs2e_c(-130, scStartTime.toLatin1().data(), &obsStartTime);
+  try {
+    Camera *cam; 
+    cam = iCube->camera();
+    cam->SetImage (0.5, 0.5);
+    iTime startTime(cam->time());
+    cam->setTime(startTime);
+    sunDist = cam->sunToBodyDist() / KM_PER_AU;
+  }
+  catch(IException &e) {
+    try {
+      loadNaifTiming();
+      sunDist = 1.0;
 
+      //  Determine if the target is a valid NAIF target
+      SpiceInt tcode;
+      SpiceBoolean found;
+      bodn2c_c(target.toLatin1().data(), &tcode, &found);
 
+      if (!found) return false;
 
-  //  Get the vector from target to sun and determine its length
-  double sunv[3];
-  double lt;
-  spkpos_c(target.toLatin1().data(), obsStartTime, "J2000", "LT+S", "sun",
-                  sunv, &lt);
-  NaifStatus::CheckErrors();
+      //  Convert starttime to et
+      double obsStartTime;
+      scs2e_c(-130, scStartTime.toLatin1().data(), &obsStartTime);
 
-  double sunkm = vnorm_c(sunv);
+      //  Get the vector from target to sun and determine its length
+      double sunv[3];
+      double lt;
+      spkpos_c(target.toLatin1().data(), obsStartTime, "J2000", "LT+S", "sun", sunv, &lt);
+      NaifStatus::CheckErrors();
 
+      double sunkm = vnorm_c(sunv);
 
-  //  Return in AU units
-  sunDist = sunkm / 1.49597870691E8;
+      //  Return in AU units
+      sunDist = sunkm / KM_PER_AU;
+    }
+    catch (IException &e) {
```

Comment:
This message doesn't ring true. This message only happens if both methods fail to get sun dist